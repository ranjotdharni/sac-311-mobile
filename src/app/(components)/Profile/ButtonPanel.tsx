import React from 'react';
import { View, StyleSheet, Dimensions, TouchableOpacity, Text } from 'react-native';
import CustomText from '../CustomText';
import { fontGetter, global, salesforceDevelopmentSignature } from '../../../customs';
import { fontSetter } from '../../../customs';
import { globalFont } from '../../../customs';
import { Link, usePathname, router } from "expo-router";

interface ButtonProps {
  text: string;
  buttonColor: string;
  textColor: string;
  onPress: () => void; // Add onPress prop
}

function Button({ text, buttonColor, textColor, onPress }: ButtonProps) {
  return (
    <View style={{ ...styles.ButtonWrapper, backgroundColor: buttonColor }}>
      <TouchableOpacity style={styles.TouchableWrapper} onPress={onPress}>
        <CustomText nol={0} font={fontGetter()} style={{ ...styles.ButtonText, color: textColor }} text={text} />
      </TouchableOpacity>
    </View>
  );
}

function ButtonDumb({ text, buttonColor, textColor, onPress }: ButtonProps) {
  return (
    <View style={{ ...styles.ButtonWrapper, backgroundColor: buttonColor }}>
      <TouchableOpacity style={styles.TouchableWrapper} onPress={onPress}>
        <Text style={{ ...styles.ButtonText, color: textColor }}>{text}</Text>
      </TouchableOpacity>
    </View>
  );
}

export default function ButtonPanel({ onPressLoginSignup }: { onPressLoginSignup: () => void },) {
  return (
    <View style={styles.ButtonPanelWrapper}>
      <Button text="Login or Signup" buttonColor="white" textColor="grey" onPress={onPressLoginSignup} />
      <ButtonDumb
        text="Submit New Request"
        buttonColor={global.baseBlue100}
        textColor="white"
        onPress={() => {}}
      >
        <Link
          to={{
            pathname: '/request/Type',
            params: {
              Subject: 'salesforceDevelopmentSignature',
              Service_Type__c: '', // CategoryLevel1
              Sub_Service_Type__c: '', // CategoryLevel2
              Council_District__c: '', // CouncilDistrictNumber
              GIS_Street_Address__c: '', // CrossStreet
              GIS_Zip_Code__c: '', // ZIP
              Address__c: '', // Address
              GIS_System_Info__c: '311 Phone', // "<Data_Source>  <SourceLevel1>"
              GIS_Neighborhood_Name__c: '', // Neighborhood
              description: '',
              Address_Geolocation__Latitude__s: 0,
              Address_Geolocation__Longitude__s: 0,
              returnRoute: usePathname().replace('/', ''),
            },
          }}
          style={styles.NewRequestLink}
        >
          <Text style={styles.requestText}>Submit New Request</Text>
        </Link>
      </ButtonDumb>
    </View>
  );
}

const styles = StyleSheet.create({
    ButtonPanelWrapper: {
        width: Dimensions.get('screen').width,
        height: Dimensions.get('screen').height * 0.22,
        backgroundColor: 'transparent', // Change this line
        position: 'absolute',
        bottom: 110,
        alignItems: 'center', // added this line to center the buttons
    },
    
    ButtonWrapper: {
        width: '92.5%',
        height: '20%',
        marginVertical: '3%',
        borderRadius: 10,
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: 'transparent',
        shadowColor: "#000", 
        shadowOffset: { 
            width: -2,
            height: 2,
        },
        shadowOpacity: 0.25, 
        shadowRadius: 3.84, 
        elevation: 7.5, 
    },
    
    TouchableWrapper: {
        width: '100%',
        height: '100%',
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: '#0000', // Add this line
    },     

    ButtonText: {
        fontSize: Dimensions.get('screen').width * 0.045,
        fontFamily: globalFont.chosenFont
    },
})