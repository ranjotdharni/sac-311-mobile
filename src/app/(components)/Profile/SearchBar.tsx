import { FontAwesome } from '@expo/vector-icons';
import React, { memo, useCallback, useState } from 'react';
import { View, TextInput, StyleSheet, Text, Image, Dimensions, TouchableOpacity } from 'react-native';
import { TouchableWithoutFeedback } from 'react-native-gesture-handler';
import { global } from '../../../customs';

function SearchBar({ placeholder, value, passUp, onSubmit, onClear, style } : { placeholder: string, value: string, passUp: (inputText: string) => void, onSubmit: () => void, style?: any, onClear: () => void }) {
    return (
        <View style={[styles.container, style ? style : {}]}>
            <View style={styles.searchInputWrapper}>
                <Image style={styles.searchIcon} source={require('../../../assets/png/search.png')} />
                {value === '' && <Text style={styles.placeholder}>{placeholder}</Text>}
                <TextInput
                    onSubmitEditing={onSubmit}
                    style={styles.searchInput}
                    value={value}
                    onChangeText={passUp}
                    placeholderTextColor="#D3D3D3"
                />
                <TouchableOpacity onPress={onClear} style={styles.clearButton}>
                    <FontAwesome name='remove' size={20} color={global.baseGrey200} />
                </TouchableOpacity>
            </View>
        </View>        
    );
}



const styles = StyleSheet.create({
    container: {
        marginTop: 10,
        backgroundColor: 'red',
        width: Dimensions.get('screen').width,
        height: 50,
        borderRadius: 25,
        overflow: 'hidden',
        display: 'flex',
        flexDirection: 'column',
        justifyContent: 'center',
        shadowColor: "#000",
        shadowOffset: {
            width: 0,
            height: 0,
        },
        shadowOpacity: 0.25,
        shadowRadius: 3.84,
    },
    searchInputWrapper: {
        width: Dimensions.get('screen').width,
        height: 50,
        borderRadius: 25, // Increase this to make the ends more circular
        elevation: 10,
        backgroundColor: 'white',
    },
    placeholder: {
        position: 'absolute', // Add this line
        left: 40, // Add this line
        top: 15, // Adjust this value as needed
        fontFamily: 'JBM', // Add this line
        color: '#D3D3D3', // Add this line
    },
    searchIcon: {
        position: 'absolute',
        left: 10,
        top: 15,
        width: 20,
        height: 20,
    },
    searchInput: {
        height: '100%',
        borderRadius: 25,
        paddingLeft: 40, // Adjust this value as needed to make room for the icon
        paddingRight: 10,
        backgroundColor: 'rgba(0, 0, 0, 0)',
        fontFamily: 'JBM',
    },
    clearButton: {
        position: 'absolute',
        width: '10%',
        height: '50%',
        left: '82.5%',
        top: '25%',
        display: 'flex',
        flexDirection: 'column',
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: global.baseBackground100,
        zIndex: 2
    },
});

export default memo(SearchBar)