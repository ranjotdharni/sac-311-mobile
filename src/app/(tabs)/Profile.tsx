import React from 'react';
import { View, Text, StyleSheet, Image, TouchableOpacity } from 'react-native';
import { useRouter } from 'expo-router';
import { globalFont, globalColorTheme } from '../../customs';

export default function Profile() {
  const router = useRouter();

  const navigateToProfile2 = () => {
    (router.push('./Profile2'));
  };

  return (
    <View style={{ flex: 1 }}>
      <Image source={require('src/assets/png/icon.png')} style={[styles.headerImage, {backgroundColor:globalColorTheme.blue}]} />
      <View style={[styles.container, {backgroundColor:globalColorTheme.backgroundColor}]}>
        <Text style={[styles.loginText, {color:globalColorTheme.color}]}>Login/Signup</Text>
        <View style={[styles.textWrapper, {backgroundColor:globalColorTheme.backgroundColor}]}>
          <Text style={[styles.createAccountText, {color:globalColorTheme.color}]}>
            Hey! Want in? Sign up and log in with your City of Sacramento account. With an account, you can track your requests, save contact info, and follow others' updates.
          </Text>
        </View>
        <TouchableOpacity onPress={navigateToProfile2} style={[styles.button, {backgroundColor:globalColorTheme.blue}]}>
          <Text style={[styles.buttonText, {color:globalColorTheme.text}]}>Login/Signup</Text>
        </TouchableOpacity>
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    zIndex: 1,
    marginTop: 0, 
  },
  headerImage: {
    resizeMode: 'center',
    width: '100%',
    height: '10%', 
    paddingHorizontal: 50,
    paddingVertical: 100,
    marginBottom: -10,
  },
  textWrapper: {
    paddingHorizontal: 5,
    maxWidth: 500,
    marginBottom: 20,
  },
  button: {
    backgroundColor: globalColorTheme.blue,
    paddingVertical: 20,
    paddingHorizontal: 100,
    borderRadius: 10,
    marginBottom: 75,
  },
  loginText: {
    fontSize: 25,
    fontFamily: globalFont.chosenFont,
    color: globalColorTheme.color,
    textAlign: 'center',
    marginBottom: 25,
  },
  buttonText: {
    fontSize: 20,
    fontFamily: globalFont.chosenFont,
    color: 'white',
    textAlign: 'center',
  },
  createAccountText: {
    fontSize: 18,
    fontFamily: globalFont.chosenFont,
    color: globalColorTheme.color,
    textAlign: 'center',
    textShadowColor: 'white',
    textShadowRadius: 1,
    marginBottom: 150,
  },
});
