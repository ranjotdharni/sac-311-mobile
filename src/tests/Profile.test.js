import renderer from "react-test-renderer";
import Profile from "../app/(tabs)/Profile";

describe("Profile", () => {
  it("renders correctly", () => {
    const page = renderer.create(<Profile />).toJSON();
    expect(page).toMatchSnapshot();
  });

  it("renders 2 children components", () => {
    const tree = renderer.create(<Profile />).toJSON();
    expect(tree.children.length).toBe(2);
  });
});