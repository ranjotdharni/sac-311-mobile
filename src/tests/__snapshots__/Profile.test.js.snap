// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile renders correctly 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <Image
    source={1}
    style={
      [
        {
          "height": "10%",
          "marginBottom": -10,
          "paddingHorizontal": 50,
          "paddingVertical": 100,
          "resizeMode": "center",
          "width": "100%",
        },
        {
          "backgroundColor": "#203d61",
        },
      ]
    }
  />
  <View
    style={
      [
        {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "marginTop": 0,
          "zIndex": 1,
        },
        {
          "backgroundColor": "#ffffff",
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#000000",
            "fontFamily": "jbm",
            "fontSize": 25,
            "marginBottom": 25,
            "textAlign": "center",
          },
          {
            "color": "#000000",
          },
        ]
      }
    >
      Login/Signup
    </Text>
    <View
      style={
        [
          {
            "marginBottom": 20,
            "maxWidth": 500,
            "paddingHorizontal": 5,
          },
          {
            "backgroundColor": "#ffffff",
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "jbm",
              "fontSize": 18,
              "marginBottom": 150,
              "textAlign": "center",
              "textShadowColor": "white",
              "textShadowRadius": 1,
            },
            {
              "color": "#000000",
            },
          ]
        }
      >
        Hey! Want in? Sign up and log in with your City of Sacramento account. With an account, you can track your requests, save contact info, and follow others' updates.
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#203d61",
          "borderRadius": 10,
          "marginBottom": 75,
          "opacity": 1,
          "paddingHorizontal": 100,
          "paddingVertical": 20,
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "white",
              "fontFamily": "jbm",
              "fontSize": 20,
              "textAlign": "center",
            },
            {
              "color": "#E7EAED",
            },
          ]
        }
      >
        Login/Signup
      </Text>
    </View>
  </View>
</View>
`;
